src/main/java/org/subethamail/smtp/server/Session.java

```diff
@@ -187,10 +187,7 @@ public void run() {
             }
             if (e instanceof RuntimeException)
                 throw (RuntimeException) e;
-            else if (e instanceof Error)
-                throw (Error) e;
-            else
-                throw new RuntimeException("Unexpected exception", e);
+            else throw (Error) e;
```

Comment:
I can't tell from the source but this is a bit worrying. e is a Throwable so can be a normal Exception which would fail cast to Error. If this is not possible that's fine but is not possible for me to check without IDE which I'll do after merge.